src/Http/Http.Results/src/ResultsCache.StatusCodes.cs

```diff
@@ -0,0 +1,154 @@
+    private static StatusCodeHttpResult? _status400BadRequest;
+    private static StatusCodeHttpResult? _status401Unauthorized;
+    private static StatusCodeHttpResult? _status402PaymentRequired;
+    private static StatusCodeHttpResult? _status403Forbidden;
+    private static StatusCodeHttpResult? _status404NotFound;
+    private static StatusCodeHttpResult? _status405MethodNotAllowed;
+    private static StatusCodeHttpResult? _status406NotAcceptable;
+    private static StatusCodeHttpResult? _status407ProxyAuthenticationRequired;
+    private static StatusCodeHttpResult? _status408RequestTimeout;
+    private static StatusCodeHttpResult? _status409Conflict;
+    private static StatusCodeHttpResult? _status410Gone;
+    private static StatusCodeHttpResult? _status411LengthRequired;
+    private static StatusCodeHttpResult? _status412PreconditionFailed;
+    private static StatusCodeHttpResult? _status413RequestEntityTooLarge;
+    private static StatusCodeHttpResult? _status414RequestUriTooLong;
+    private static StatusCodeHttpResult? _status415UnsupportedMediaType;
+    private static StatusCodeHttpResult? _status416RequestedRangeNotSatisfiable;
+    private static StatusCodeHttpResult? _status417ExpectationFailed;
+    private static StatusCodeHttpResult? _status418ImATeapot;
+    private static StatusCodeHttpResult? _status419AuthenticationTimeout;
+    private static StatusCodeHttpResult? _status421MisdirectedRequest;
+    private static StatusCodeHttpResult? _status422UnprocessableEntity;
+    private static StatusCodeHttpResult? _status423Locked;
+    private static StatusCodeHttpResult? _status424FailedDependency;
+    private static StatusCodeHttpResult? _status426UpgradeRequired;
+    private static StatusCodeHttpResult? _status428PreconditionRequired;
+    private static StatusCodeHttpResult? _status429TooManyRequests;
+    private static StatusCodeHttpResult? _status431RequestHeaderFieldsTooLarge;
+    private static StatusCodeHttpResult? _status451UnavailableForLegalReasons;
+    private static StatusCodeHttpResult? _status500InternalServerError;
+    private static StatusCodeHttpResult? _status501NotImplemented;
+    private static StatusCodeHttpResult? _status502BadGateway;
+    private static StatusCodeHttpResult? _status503ServiceUnavailable;
+    private static StatusCodeHttpResult? _status504GatewayTimeout;
+    private static StatusCodeHttpResult? _status505HttpVersionNotsupported;
+    private static StatusCodeHttpResult? _status506VariantAlsoNegotiates;
+    private static StatusCodeHttpResult? _status507InsufficientStorage;
+    private static StatusCodeHttpResult? _status508LoopDetected;
+    private static StatusCodeHttpResult? _status510NotExtended;
+    private static StatusCodeHttpResult? _status511NetworkAuthenticationRequired;
+
+    internal static StatusCodeHttpResult StatusCode(int statusCode)
+    {
+        if ( statusCode is (< 100) or (> 511))
```

Comment:
![PerfectionFlawlessGIF](https://user-images.githubusercontent.com/95136/160952643-e1ef31d6-7090-422d-a625-830ab6d5aad8.gif)

Reply:
For reference: Perf-goodness will come for free with https://github.com/dotnet/roslyn/issues/60534